app/forms/finance/ecf/change_training_status_form.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module Finance
+  module ECF
+    class ChangeTrainingStatusForm
+      include ActiveModel::Model
+      include ActiveModel::Attributes
+
+      REASON_OPTIONS = {
+        "deferred" => Participants::Defer::ECF.reasons,
+        "withdrawn" => Participants::Withdraw::ECF.reasons,
+      }.freeze
+
+      attribute :participant_profile
+      attribute :training_status
+      attribute :reason
+
+      validates :training_status, inclusion: ParticipantProfile.training_statuses.values
+      validates :reason, inclusion: { in: :valid_training_status_reasons }, if: :reason_required?
+
+      def training_status_options
+        ParticipantProfile.training_statuses.except(current_training_status)
+      end
+
+      def reason_required?
+        training_status.present? && training_status != "active"
+      end
+
+      def reason_options
+        REASON_OPTIONS.except(current_training_status)
+      end
+
+      def current_training_status
+        induction_record&.training_status
+      end
+
+      def valid_training_status_reasons
+        reason_options[training_status] || []
```

Comment:
should not need this `|| []` because reason require will only be `true` if the `training_status` is not active. hence reason option will always return something an array. If this breaks than let the error bubble up and we'll address it if it ever happens. Otherwise we may never notice that there is something odd happening